roofit/roofitcore/src/RooConstraintSum.cxx

```diff
@@ -93,6 +93,28 @@ double RooConstraintSum::evaluate() const
 }
 
 
+void RooConstraintSum::computeBatch(cudaStream_t *, double *output, size_t /*size*/,
+                                    RooFit::Detail::DataMap const &dataMap) const
+{
+   double sum(0);
+
+   for (const auto comp : _set1) {
+      sum -= std::log(dataMap.at(comp)[0]);
+   }
+
+   output[0] = sum;
```

Comment:
*Maybe* a KahanSum could make sense here. The probabilities should often close to one, but they might fall rapidly when NPs get pulled. This could effectively disable a part of the sum if you use only `-=`.
What do you think?

Reply:
Yes makes sense, after all the Kahan summation is also used for the components on the NLL so it would be more consistent. I'll update this!

Reply:
But better in a different PR, because then I also want to change this in `evaluate()`, and this PR should only touch the batch mode